llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -1705,9 +1711,10 @@ static void computeKnownBitsFromOperator(const Operator *I,
                     m_Br(m_c_ICmp(Pred, m_Specific(IncValue), m_APInt(RHSC)),
```

Comment:
Can we fix this by replacing RecQ.CxtI with `P->getIncomingBlock(u)->getTerminator()` instead? (In terms of minimally ugly correct implementation, not maximally theoretically powerful.)

Reply:
I don't think so. Take the test case. If the branch in `%B1` used a `%cmp` on `%sub` that would be patently incorrect to use when analyzing `%sub` in `%B0`